gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GcsJavaClientImpl.java

```diff
@@ -29,22 +37,35 @@ public class GcsJavaClientImpl implements GoogleCloudStorage {
   private GoogleCloudStorageImpl gcsClientDelegate;
 
   private GoogleCloudStorageOptions storageOptions;
-  private Credentials credentials;
+  private Storage storage;
+
+  // Thread-pool used for background tasks.
+  private final ExecutorService backgroundTasksThreadPool =
+      Executors.newCachedThreadPool(
+          new ThreadFactoryBuilder()
+              .setNameFormat("gcs-java-client-write-channel-pool-%d")
+              .setDaemon(true)
+              .build());
 
   private GcsJavaClientImpl(GcsJavaClientImplBuilder builder) throws IOException {
     this.storageOptions = checkNotNull(builder.storageOptions, "options must not be null");
-    this.credentials = checkNotNull(builder.credentials, "credentials must not be null");
-
+    this.storage = checkNotNull(builder.javaClientStorage, "storage must not be null");
```

Comment:
As discussed offline. It's the `storage` from `google-cloud-storage` library. Unlike the one from Apiary It can't be faked via HttpTransport and needed to be faked/mocked by passing as argument only.